lua/luasnip/extras/postfix.lua

```diff
@@ -2,53 +2,61 @@ local snip = require("luasnip.nodes.snippet").S
 local events = require("luasnip.util.events")
 local extend_decorator = require("luasnip.util.extend_decorator")
 local node_util = require("luasnip.nodes.util")
+local util = require("luasnip.util.util")
 
 local matches = {
 	default = [[[%w%.%_%-%"%']+$]],
 	line = "^.+$",
 }
 
-local function generate_opts(match_pattern, user_callback)
+local function wrap_resolve_expand_params(match_pattern, user_resolve)
+	return function(snippet, line_to_cursor, match, captures)
+		if snippet.condition then
+			if not snippet.condition(line_to_cursor, match, captures) then
+				return nil
+			end
+		end
+
+		local pos = util.get_cursor_0ind()
+		local line_to_cursor_except_match =
+			line_to_cursor:sub(1, #line_to_cursor - #match)
+		local postfix_match = line_to_cursor_except_match:match(match_pattern)
+			or ""
+		local res = {
+			trigger = match,
+			captures = captures,
+			clear_region = {
+				from = { pos[1], pos[2] - #postfix_match - #match },
+				to = pos,
+			},
+			env_override = {
+				POSTFIX_MATCH = postfix_match,
+			},
```

Comment:
I've looked into the test-failures: Looks like `POSTFIX_MATCH` is not passed all the way into the snippet, does that work on your machine?

Reply:
i'm trying fix this. in postfix testcases, `snip_expand` is invoked directly, skip ` matches`, so the `POSTFIX_MATCH` would not be set. qaq

Reply:
Ahh right, good catch.
I can handle the test-cases if you like :)

Reply:
fix it, by adding a new func `snip_try_expand`. take a look :)